src/lightning_app/testing/testing.py

```diff
@@ -137,9 +137,12 @@ def application_testing(
 
     from click.testing import CliRunner
 
-    with mock.patch("lightning.LightningApp", lightning_app_cls), mock.patch(
-        "lightning_app.LightningApp", lightning_app_cls
-    ):
+    patch1 = mock.patch("lightning_app.LightningApp", lightning_app_cls)
+    # we need to patch both only with the mirror package
+    patch2 = (
+        mock.patch("lightning.LightningApp", lightning_app_cls) if "lightning.app" in sys.modules else nullcontext()
+    )
+    with patch1, patch2:
```

Comment:
```suggestion
    _patch1 = mock.patch("lightning_app.LightningApp", lightning_app_cls)
    # we need to patch both only with the mirror package
    _patch2 = (
        mock.patch("lightning.LightningApp", lightning_app_cls) if "lightning.app" in sys.modules else nullcontext()
    )
    with _patch1, _patch2:
```

Reply:
Why this suggestion? These are local variables in a function

Reply:
ok I missed as it, too zoom code :D